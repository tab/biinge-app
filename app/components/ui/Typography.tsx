import React from "react"
import { Text, StyleProp, TextStyle } from "react-native"

import { typographyStyles, TypographyVariantType } from "styles"

type Props = {
  style?: StyleProp<TextStyle>
  variant: TypographyVariantType
  numberOfLines?: number
  children: React.ReactNode
}

const TypographyComponent = ({
  style,
  variant,
  numberOfLines,
  children,
}: Props) => {
  const getVariantStyle = () => typographyStyles[variant] || {}

  return (
    <Text numberOfLines={numberOfLines} style={[getVariantStyle(), style]}>
      {children}
    </Text>
  )
}

TypographyComponent.defaultProps = {
  numberOfLines: 0,
}

export default TypographyComponent
