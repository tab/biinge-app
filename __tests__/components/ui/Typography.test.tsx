import React from "react"
import "react-native"
import { render } from "@testing-library/react-native"
import { it } from "@jest/globals"

import Typography from "components/ui/Typography"

describe("Typography", () => {
  const title = "Title"

  it("renders correctly title1 variant", () => {
    const tree = render(<Typography variant="title1">{title}</Typography>)
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly title2 variant", () => {
    const tree = render(<Typography variant="title2">{title}</Typography>)
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly title3 variant", () => {
    const tree = render(<Typography variant="title3">{title}</Typography>)
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly headline variant", () => {
    const tree = render(<Typography variant="headline">{title}</Typography>)
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly body variant", () => {
    const tree = render(<Typography variant="body">{title}</Typography>)
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly callout variant", () => {
    const tree = render(<Typography variant="callout">{title}</Typography>)
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly subhead variant", () => {
    const tree = render(<Typography variant="subhead">{title}</Typography>)
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly footnote variant", () => {
    const tree = render(<Typography variant="footnote">{title}</Typography>)
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly caption1 variant", () => {
    const tree = render(<Typography variant="caption1">{title}</Typography>)
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly caption2 variant", () => {
    const tree = render(<Typography variant="caption2">{title}</Typography>)
    expect(tree).toMatchSnapshot()
  })
})
