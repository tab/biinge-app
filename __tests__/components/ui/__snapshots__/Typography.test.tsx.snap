// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography renders correctly body variant 1`] = `
<Text
  ellipsizeMode="tail"
  numberOfLines={0}
  style={
    [
      {
        "color": "#181818",
        "fontSize": 18,
        "fontWeight": "300",
        "lineHeight": 22,
      },
      undefined,
    ]
  }
>
  Title
</Text>
`;

exports[`Typography renders correctly callout variant 1`] = `
<Text
  ellipsizeMode="tail"
  numberOfLines={0}
  style={
    [
      {
        "color": "#181818",
        "fontSize": 18,
        "fontWeight": "600",
        "lineHeight": 24,
      },
      undefined,
    ]
  }
>
  Title
</Text>
`;

exports[`Typography renders correctly caption1 variant 1`] = `
<Text
  ellipsizeMode="tail"
  numberOfLines={0}
  style={
    [
      {
        "color": "#181818",
        "fontSize": 14,
        "fontWeight": "700",
        "lineHeight": 17,
      },
      undefined,
    ]
  }
>
  Title
</Text>
`;

exports[`Typography renders correctly caption2 variant 1`] = `
<Text
  ellipsizeMode="tail"
  numberOfLines={0}
  style={
    [
      {
        "color": "#181818",
        "fontSize": 14,
        "fontWeight": "600",
        "lineHeight": 17,
      },
      undefined,
    ]
  }
>
  Title
</Text>
`;

exports[`Typography renders correctly footnote variant 1`] = `
<Text
  ellipsizeMode="tail"
  numberOfLines={0}
  style={
    [
      {
        "color": "#181818",
        "fontSize": 13,
        "fontWeight": "400",
        "lineHeight": 18,
      },
      undefined,
    ]
  }
>
  Title
</Text>
`;

exports[`Typography renders correctly headline variant 1`] = `
<Text
  ellipsizeMode="tail"
  numberOfLines={0}
  style={
    [
      {
        "color": "#181818",
        "fontSize": 17,
        "fontWeight": "600",
        "lineHeight": 22,
      },
      undefined,
    ]
  }
>
  Title
</Text>
`;

exports[`Typography renders correctly subhead variant 1`] = `
<Text
  ellipsizeMode="tail"
  numberOfLines={0}
  style={
    [
      {
        "color": "#181818",
        "fontSize": 16,
        "fontWeight": "700",
        "lineHeight": 20,
      },
      undefined,
    ]
  }
>
  Title
</Text>
`;

exports[`Typography renders correctly title1 variant 1`] = `
<Text
  ellipsizeMode="tail"
  numberOfLines={0}
  style={
    [
      {
        "color": "#181818",
        "fontSize": 26,
        "fontWeight": "700",
        "lineHeight": 28,
      },
      undefined,
    ]
  }
>
  Title
</Text>
`;

exports[`Typography renders correctly title2 variant 1`] = `
<Text
  ellipsizeMode="tail"
  numberOfLines={0}
  style={
    [
      {
        "color": "#181818",
        "fontSize": 22,
        "fontWeight": "400",
        "lineHeight": 28,
      },
      undefined,
    ]
  }
>
  Title
</Text>
`;

exports[`Typography renders correctly title3 variant 1`] = `
<Text
  ellipsizeMode="tail"
  numberOfLines={0}
  style={
    [
      {
        "color": "#181818",
        "fontSize": 20,
        "fontWeight": "400",
        "lineHeight": 25,
      },
      undefined,
    ]
  }
>
  Title
</Text>
`;
